cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,245 @@
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods --for [<type>[.<api-group>]/]<name>",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFilterFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	if !o.PrintFlags.OutputFlagSpecified() {
+		o.PrintObj = printPodTable
+	} else {
+		printer, err := o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.PrintObj = printer.PrintObj
+	}
+
+	o.ForObject, err = parseForObjectFilterFlag(o.UserSpecifiedForObject)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	namespace := o.Namespace
+	if o.AllNamespaces {
+		namespace = ""
+	}
+
+	continueToken := ""
+	filteredPodListSlice := make([]*corev1.PodList, 0)
+	for {
+		podList, err := o.Clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: o.LabelSelector,
+			FieldSelector: o.FieldSelector,
+			Continue:      continueToken,
+		})
+		if err != nil {
+			return err
+		}
+
+		if fp := o.filterPodsByOwnerRef(podList); len(fp) != 0 {
+			podList.Items = fp
+			filteredPodListSlice = append(filteredPodListSlice, podList)
+		}
+
+		if podList.Continue == "" {
+			break
+		}
+		continueToken = podList.Continue
+	}
+
+	// handle if no filtered podList found
+	if len(filteredPodListSlice) == 0 {
+		if !o.AllNamespaces {
+			fmt.Fprintf(o.ErrOut, "No resources found in %s namespace.\n", o.Namespace)
+		} else {
+			fmt.Fprintln(o.ErrOut, "No resources found.")
+		}
+		return nil
+	}
+
+	// now range over the filteredPodListSlice and print them
+	for _, filteredPodList := range filteredPodListSlice {
```

Comment:
I think in this case it will show column names in each iteration.

Reply:
Done https://github.com/kubernetes-sigs/kueue/pull/2280/commits/b5a12206958932fcdbedcac55aaeeb16a7b19b3a

Reply:
Thanks.